docs/how-to/react-server-components.md

```diff
@@ -1,50 +1,738 @@
+    // The app routes.
+    routes: routes(),
+    // Encode the match with the React Server implementation.
+    generateResponse(match) {
+      return new Response(renderToReadableStream(match.payload), {
+        status: match.statusCode,
+        headers: match.headers,
+      });
+    },
+  });
+}
+
+const app = express();
+
+// Serve static assets with compression and long cache lifetime.
+app.use(
+  "/client",
+  compression(),
+  express.static("dist/client", {
+    immutable: true,
+    maxAge: "1y",
+  })
+);
+// Hook up our application.
+app.use(
+  createRequestListener((request) =>
+    generateHTML(
+      request,
+      fetchServer,
+      (routes as unknown as { bootstrapScript?: string }).bootstrapScript
+    )
+  )
+);
+
+app.listen(3000, () => {
+  console.log("Server listening on port 3000");
+});
+```
+
+#### `entry.client.tsx`
+
+```tsx filename=src/entry.client.tsx
+"use client-entry";
+
+import { startTransition, StrictMode } from "react";
+import { hydrateRoot } from "react-dom/client";
+import {
+  unstable_createCallServer as createCallServer,
+  unstable_getRSCStream as getRSCStream,
+  unstable_RSCHydratedRouter as RSCHydratedRouter,
+  type unstable_RSCPayload as RSCServerPayload,
+} from "react-router";
+import {
+  createFromReadableStream,
+  createTemporaryReferenceSet,
+  encodeReply,
+  setServerCallback,
+} from "react-server-dom-parcel/client";
+
+// Create and set the callServer function to support post-hydration server actions.
+setServerCallback(
+  createCallServer({
+    createFromReadableStream,
+    createTemporaryReferenceSet,
+    encodeReply,
+  })
+);
+
+// Get and decode the initial server payload.
+createFromReadableStream(getRSCStream()).then(
+  (payload: RSCServerPayload) => {
+    startTransition(async () => {
+      const formState =
+        payload.type === "render"
+          ? await payload.formState
+          : undefined;
+
+      hydrateRoot(
+        document,
+        <StrictMode>
+          <RSCHydratedRouter
+            createFromReadableStream={
+              createFromReadableStream
+            }
+            payload={payload}
+          />
+        </StrictMode>,
+        {
+          formState,
+        }
+      );
+    });
+  }
+);
+```
+
+### Vite
+
+See the [Vite RSC docs][vite-rsc-doc] for more information.
+
+In addition to `react`, `react-dom`, and `react-router`, you'll need the following dependencies:
+
+```shellscript
+npm i -D vite @vitejs/plugin-react @vitejs/plugin-rsc
+```
+
+#### `vite.config.ts`
+
+To configure Vite, add the following to your `vite.config.ts`:
+
+```ts filename=vite.config.ts
+import rsc from "@vitejs/plugin-rsc/plugin";
+import react from "@vitejs/plugin-react";
+import { defineConfig } from "vite";
+
+export default defineConfig({
+  plugins: [
+    react(),
+    rsc({
+      entries: {
+        client: "src/entry.client.tsx",
+        rsc: "src/entry.rsc.tsx",
+        ssr: "src/entry.ssr.tsx",
+      },
+    }),
+  ],
+});
+```
+
```

Comment:
We're missing a route config here similar to the Parcel docs above.

Reply:
Ah yeah, for completion sake that's probably worth adding